src/goto-instrument/contracts/contracts.cpp

```diff
@@ -1175,6 +1175,10 @@ void code_contractst::check_frame_conditions(
     else if(instruction_it->is_dead())
     {
       const auto &symbol = instruction_it->dead_symbol();
+      source_locationt &location_no_checks =
+        instruction_it->source_location_nonconst();
+      disable_pointer_checks(location_no_checks);
```

Comment:
As above: is the reference intentional?

Reply:
fixed

Reply:
We could add a comment here again explaining "why" is safe to always remove pointer checks in this location.